pydantic/env_settings.py

```diff
@@ -36,23 +38,34 @@ def _substitute_environ(self):
         Substitute environment variables into values.
         """
         d = {}
+
+        if self.__config__.case_insensitive:
+            env_vars = {k.lower(): v for (k, v) in os.environ.items()}
+        else:
+            env_vars = os.environ
+
         for field in self.__fields__.values():
+
             if field.has_alias:
                 env_name = field.alias
             else:
                 env_name = self.__config__.env_prefix + field.name.upper()
-            env_var = os.getenv(env_name, None)
-            if env_var:
+
+            env_name_ = env_name.lower() if self.__config__.case_insensitive else env_name
+            env_val = env_vars.get(env_name_, None)
```

Comment:
I came to the conclusion that `env_val` was a better ident name now that we have `env_vars` above. `env_var` in context of `env_vars` sounds like an item ident. Buts its not. Its a value from the environment while `env_vars` is the environment (dict).

if you prefer to not touch `env_var` ident name, I would suggest using a different name than `env_vars` above. Maybe `normalized_environs`?